src/contexts/FavouriteMessageContext.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { createContext, useEffect, useReducer } from "react";
+
+import { Action } from "../dispatcher/actions";
+
+let cachedFavouriteMessageIds: string[] = [];
+
+const ActionHandler = (_, action: {type: Action, eventId: string}) => {
+    switch (action.type) {
+        case Action.OnAddToFavourite:
+            cachedFavouriteMessageIds.push(action.eventId);
+            break;
+        case Action.OnRemoveFromFavourite:
+            cachedFavouriteMessageIds.splice(cachedFavouriteMessageIds.indexOf(action.eventId), 1);
+            break;
+        default:
+            throw new Error("Ensure a valid Action is dispatched");
+    }
+    localStorage.setItem('io_element_favouriteMessages', JSON.stringify(cachedFavouriteMessageIds));
+    return [...cachedFavouriteMessageIds];
+};
+
+type TContext = {
+    favouriteMessageIds: string[];
+    dispatch: React.Dispatch<any>;
+};
+export const FavouriteMessageContext = createContext<TContext | null>({
+    favouriteMessageIds: [],
+    dispatch: null,
+});
+
+const FavouriteMessageProvider = ({ children }) => {
+    const [favouriteMessageIds, dispatch] = useReducer(ActionHandler,
```

Comment:
I tried to use that once but it fails me, i need to try again today. Thanks

Reply:
@SimonBrandner Can i get a suggestion on how i might use the `MatrixDispatcher ` in my use case since i am using Context, Reducers etc. I tried following the existing approach of how element uses it's dispatch, it's not firing once i click the button that should trigger the dispatch. I hope you understand me. Also is there any advantage of using the `MatrixDispatcher ` over the default React Dispatcher in this case

Reply:
So I am not sure using a context or `MatrixDispatcher` here makes much sense. IMO, the simplest solution would be to add a custom hook (e.g. `useFavouriteMessages`) which would return an array of the favourited messages and functions to add and remove them

<hr>

Take this with a grain of sand - it might make sense for someone with more experience to take a look; let's not block the PR on this

Reply:
@t3chguy please can you review this change, and especially comment on the use of the React dispatcher - is this ok?